File: egghead.io/data-structures-and-algorithms-in-js/lesson-11-sort-an-arr-with-a-js-do-while-loop-using-bubble-sort/bubble_sort.py
def bubble_sort(array):
  while True:
    is_swapped = False

    for i in range(len(array) - 1):
      print(array)
      
      if array[i] > array[i + 1]:
        curr = array[i]
        array[i] = array[i + 1]
        array[i + 1] = curr
        is_swapped = True

    if not is_swapped: break

  return array

File: egghead.io/data-structures-and-algorithms-in-js/lesson-11-sort-an-arr-with-a-js-do-while-loop-using-bubble-sort/bubble-sort.js
const bubbleSort = (array) => {
  let isSwapped;

  do {
    isSwapped = false;

    array.forEach((value, index, array) => {
      console.log(array);

      if (value > array[index + 1]) {
        const curr = value;
        array[index] = array[index + 1];
        array[index + 1] = curr;
        isSwapped = true; 
      }
    });

  } while (isSwapped);

  return array;
}

module.exports = bubbleSort;
